src/Umbraco.Core/Install/FilePermissionDirectoryHelper.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.IO;
+using Umbraco.Core.IO;
+
+namespace Umbraco.Web.Install
+{
+    public class FilePermissionDirectoryHelper
```

Comment:
Try to avoid "Helper" class names, maybe this can just be IOHelper extension methods? depending on if CreateRandomFileName is used outside of this class.

Reply:
I moved both methods into extensions of `IIOHelper`. Even that only one of them uses `IIOHelper`. The reason to move the other method also (`CreateRandomFileName()`) is because I think it logically belongs to `IOHelper`, but we don't need to implement this method for every implementation.